cmd/provider/main.go

```diff
@@ -78,8 +80,9 @@ func main() {
 		Options: xpcontroller.Options{
 			Logger:                  log,
 			GlobalRateLimiter:       ratelimiter.NewGlobal(*maxReconcileRate),
-			PollInterval:            1 * time.Minute,
-			MaxConcurrentReconciles: 1,
+			PollInterval:            *pollInterval,
+			MaxConcurrentReconciles: *maxReconcileRate,
+			Features:                &feature.Flags{},
 		},
 		Provider: config.GetProvider(),
 		// use the following WorkspaceStoreOption to enable the shared gRPC mode
```

Comment:
To enable the `initProvider` we need to pass the features flags to the WorkspaceStore. Check https://github.com/upbound/provider-gcp/pull/336/commits/131c585d6c8a944eb8ab7d3387f75c4942a3fae0, for details

Reply:
@lsviben thanks for review add  https://github.com/upbound/provider-vault/pull/7/commits/9096797eea699a21e794fd3bd19f3933e8de52be